packages/core/src/utils/breakpoints.ts

```diff
@@ -49,21 +41,14 @@ export const mediaQueryMatcher = (breakpoints: Breakpoint[]) => {
 };
 
 export const getActiveBreakpointIndex = (
-  breakpoints: Breakpoint[],
   mediaQueryMatches: Record<string, boolean>,
   fallbackBreakpointIndex: number
 ) => {
-  // The breakpoints are ordered (desktop-first: descending by screen width)
-  const breakpointsWithMatches = breakpoints.map(({ id }, index) => ({
-    id,
-    index,
-    // The fallback breakpoint with wildcard query will always match
-    isMatch: mediaQueryMatches[id] ?? true,
-  }));
-
-  // Find the last breakpoint in the list that matches (desktop-first: the narrowest one)
-  const mostSpecificIndex = findLast(breakpointsWithMatches, ({ isMatch }) => isMatch)?.index;
-  return mostSpecificIndex ?? fallbackBreakpointIndex;
+  if (Object.keys(mediaQueryMatches).length === 0) return fallbackBreakpointIndex;
+  if (mediaQueryMatches.mobile) return 2;
+  if (mediaQueryMatches.tablet) return 1;
+  if (mediaQueryMatches.desktop) return 0;
```

Comment:
We wrote the breakpoint logic so that it would allow a dynamic number instead of solely the three default ones. It would be great if we didn't remove this functionality as we would have to re-introduce it again later.

Reply:
Also, note that your current logic doesn't care about whether it's `desktop-first` vs. `mobile-first`. It looks like this introduces yet another bug :/

Was any change in this file necessary for the fix? It looks like the actual fix only happened in `useBreakpoints`. If so, I would ask you to leave the logic in here unchanged to keep it adjustable to different breakpoint definitions :)

Reply:
Hmm for some reason when I was testing it on Friday this logic was giving the wrong output, but now I did a fresh build of everything, it seems to work as expected. So I will revert this